app/src/androidTest/java/com/github/sdp_begreen/begreen/fragments/CameraWithUIFragmentTest.kt

```diff
@@ -0,0 +1,172 @@
+package com.github.sdp_begreen.begreen.fragments
+
+import android.Manifest
+import androidx.fragment.app.testing.FragmentScenario
+import androidx.fragment.app.testing.launchFragmentInContainer
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.action.ViewActions
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.matcher.RootMatchers
+import androidx.test.espresso.matcher.ViewMatchers
+import androidx.test.espresso.matcher.ViewMatchers.assertThat
+import androidx.test.ext.junit.rules.ActivityScenarioRule
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.filters.LargeTest
+import androidx.test.rule.GrantPermissionRule
+import com.github.sdp_begreen.begreen.R
+import com.github.sdp_begreen.begreen.activities.MainActivity
+import com.github.sdp_begreen.begreen.firebase.Auth
+import com.github.sdp_begreen.begreen.firebase.DB
+import com.github.sdp_begreen.begreen.models.User
+import com.github.sdp_begreen.begreen.rules.KoinTestRule
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.test.runTest
+import org.hamcrest.CoreMatchers.instanceOf
+import org.junit.Before
+import org.junit.BeforeClass
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.koin.dsl.module
+import org.mockito.Mockito
+
+@RunWith(AndroidJUnit4::class)
+@LargeTest
+class CameraWithUIFragmentTest {
+    private lateinit var fragmentScenario: FragmentScenario<CameraWithUIFragment>
+
+    //Companion object to mock the DB and Auth
+    companion object {
+        val user = User("test", 2, "test", 5, null, "test", "test", "test", 15)
+        private val db: DB = Mockito.mock(DB::class.java)
+        private val auth: Auth = Mockito.mock(Auth::class.java)
+        val users = listOf<User>(
+            User("1", 123, "Alice"),
+            User("2", 0, "Bob Zeu bricoleur"),
+            User("3", 14, "Charlie Chaplin"),
+            User("4", 23, "David Pujadas"),
+            User("5", 10492, "Euler"),
+            User("6", 1234, "Alain Berset"),
+            User("7", 1235, "Mister Alix")
+        )
+
+        // Setup the mock
+        @OptIn(ExperimentalCoroutinesApi::class)
+        @BeforeClass
+        @JvmStatic
+        fun setUp() {
+            runTest {
+                // setup basic get user and getProfilePicture use in multiple tests
+                Mockito.`when`(db.getUser(user.id))
```

Comment:
Here if you want to import `when` statically for readability you could. As you prefer.